kubemarine/kubernetes/__init__.py

```diff
@@ -663,23 +641,39 @@ def apply_labels(group):
 
 def apply_taints(group):
     log = group.cluster.log
-
-    log.debug("Applying additional taints for nodes")
-    with RemoteExecutor(group.cluster):
-        for node in group.get_ordered_members_list(provide_node_configs=True):
-            if "taints" not in node:
-                log.verbose("No additional taints found for %s" % node['name'])
-                continue
-            log.verbose("Found additional taints for %s: %s" % (node['name'], node['taints']))
-            for taint in node["taints"]:
-                group.cluster.nodes["control-plane"].get_first_member() \
-                    .sudo("kubectl taint node %s %s" % (node["name"], taint))
+    minor_version = '.'.join((group.cluster.inventory["services"]["kubeadm"]["kubernetesVersion"]).split('.')[:-1])
+    global_taint = group.cluster.globals['taint'][minor_version]
+    worker_found = False
+    log.debug("Check taints for nodes")
+    for node in group.get_ordered_members_list(provide_node_configs=True):
+        if "worker" in node["roles"]:
+            worker_found = True
+
+            if "control-plane" in node["roles"] and worker_found:
+                result = node['connection'].sudo("kubectl get nodes %s -o=jsonpath='{range}{.spec.taints[*].key}{end}'" % (node['name']))
```

Comment:
I believe this is still possible to use RemoteExecutor, but split the commands into two batches instead of one:
* 1) Collect all taints from worker & control-plane
* 2) Apply default and custom taints in bulk.

Reply:
rework procedure